modules/lvgl/input/lvgl_common_input.c

```diff
@@ -52,3 +55,26 @@ int lvgl_input_register_driver(lv_indev_type_t indev_type, const struct device *
 
 	return 0;
 }
+
+#define LV_DEV_INIT(node_id, init_fn)                                                              \
+	do {                                                                                       \
+		int ret = init_fn(DEVICE_DT_GET(node_id));                                         \
+		if (ret) {                                                                         \
+			return ret;                                                                \
+		}                                                                                  \
+	} while (0)
+
+int lvgl_init_input_devices(void)
+{
+	int ret = 0;
+
+	DT_FOREACH_STATUS_OKAY_VARGS(zephyr_lvgl_pointer_input, LV_DEV_INIT,
+				     lvgl_pointer_input_init)
+
+	DT_FOREACH_STATUS_OKAY_VARGS(zephyr_lvgl_button_input, LV_DEV_INIT, lvgl_button_input_init)
+
+	DT_FOREACH_STATUS_OKAY_VARGS(zephyr_lvgl_encoder_input, LV_DEV_INIT,
+				     lvgl_encoder_input_init)
```

Comment:
two things:
1. these needs some `;` at the end of the line
2. unfortunately it's possible for these device to be "okay" and the driver not being build, then this would build fail, so I'm afraid these will have to be enclosed in some `#if LV_Z_POINTER_INPUT` etc...

Reply:
Sure will add that.

> unfortunately it's possible for these device to be "okay" and the driver not being build,

Is that because any of the other prerequisites (e.g. CONFIG_LVGL) not being fulfilled?

Reply:
yeah, CONFIG_INPUT I think, regardless, that's the condition that enable the file that defines the function so all good